myhomework4.php

```diff
@@ -0,0 +1,203 @@
+const FILE = "passwords";
+//function readHttpLikeInput()
+//{
+//    $f = fopen('php://stdin', 'r');
+//    $store = "";
+//    $toread = 0;
+//    while ($line = fgets($f)) {
+//        $store .= preg_replace("/\r/", "", $line);
+//        if (preg_match('/Content-Length: (\d+)/', $line, $m))
+//            $toread = $m[1] * 1;
+//        if ($line == "\r\n")
+//            break;
+//    }
+//    if ($toread > 0)
+//        $store .= fread($f, $toread);
+//    return $store;
+//}
+//
+//$contents = readHttpLikeInput();
+
+function outputHttpResponse($statuscode, $statusmessage, $headers, $body)
+{
+    $response = "HTTP/1.1 $statuscode $statusmessage" . PHP_EOL;
+    $response .= "Date: " . date("l, d F  Y h:i:sa") . PHP_EOL;
+    $response .= "Server: Apache/2.2.14 (Win32)" . PHP_EOL;
+    $response .= "Content-Length: " . strlen($body) . PHP_EOL;
+    $response .= "Connection: Closed" . PHP_EOL;
+    $response .= "Content-Type: text/html; charset=utf-8" . PHP_EOL;
+    $response .= PHP_EOL . "$body";
+
+    echo $response;
+}
+
+function processHttpRequest($method, $uri, $headers, $body)
+{
+    try{
+        $statusCode = getStatusCode($headers, $uri, $body);
+        $statusMessage = "Found";
+    }catch (Exception $e){
+        $statusCode = $e->getCode();
+        $statusMessage = $e->getMessage();
+    }
+
+    $body = getBodyMessage($statusCode,$statusMessage);
+    outputHttpResponse($statusCode, $statusMessage, $headers, $body);
+}
+
+function getBodyMessage($statuscode , $statusMessage){
+    $color = ($statuscode == HttpStatusCodes::OK) ? "green" : "red";
+
+    return '<h1 style="color:' . $color . '">' . $statusMessage . '</h1>';
+}
+
+
+function getStatusCode($headers, $uri, $body)
+{
+    if (!checkUri($uri) && !checkContType($headers)) {//якщо неправильний урі або контент тайп
+        throw new Exception("Bad Request", HttpStatusCodes::BAD_REQUEST);
+    }
+
+    if (!checkPassTxt()) {
+        throw new Exception("Internal Server Error", HttpStatusCodes::INTERNAL_SERVER_ERROR);
+    }
+
+    if (!checkLogin($body)) {
+        throw new Exception("Bad Request", HttpStatusCodes::BAD_REQUEST);
+    }
+
+    if (!checkPassword($body)) {//якщо неправильний  пароль
+        throw new Exception("Unauthorized", HttpStatusCodes::UNAUTHORIZED);
+    }
+
+    return HttpStatusCodes::OK;
+}
+
+function checkPassword($body)
+{
+    $password = getPassword($body);
+    $login = getLogin($body);
+    $data = file_get_contents(FILE);
+
+    if (str_contains($data, $login)) {
+        $firstValuePassword = explode($login . ':', $data)[1];
+        $truePassword = explode(PHP_EOL, $firstValuePassword)[0];
+
+        return $truePassword === $password;
+    }
+
+    return false;
+}
+
+
+function getPassword($body)
+{
+
+    return explode('password=', $body)[1];
+}
+
+function checkPassTxt()
+{
+
+    return file_exists(FILE);
+}
+
+function checkLogin($body)
+{
+    $data = file_get_contents(FILE);
```

Comment:
you read file twice, that's bad decision.
you may have error like `Fatal error: Allowed memory size of 94371840 bytes exhausted (tried to allocate 5072 bytes)`